halo2_proofs/src/poly/domain.rs

```diff
@@ -201,6 +202,30 @@ impl<G: Group> EvaluationDomain<G> {
         }
     }
 
+    /// Returns a  polynomial in the Lagrange coefficient basis
+    /// with an empty(zero) value in used cells and random value in unused cells
+    pub(crate) fn empty_lagrange_assigned_with_unused_random(
+        &self,
+        unusable_rows_start: usize,
+        mut rng: impl RngCore,
+    ) -> Polynomial<Assigned<G>, LagrangeCoeff>
+    where
+        G: Field,
+    {
+        Polynomial {
+            values: (0..self.n)
+                .map(|i| {
+                    if i < unusable_rows_start as u64 {
+                        G::group_zero().into()
+                    } else {
+                        G::random(&mut rng).into()
+                    }
+                })
+                .collect(),
+            _marker: PhantomData,
+        }
+    }
```

Comment:
This should initialise the polynomial to random values on all rows (not just unusable rows). Assigning a cell would overwrite the initial random value; unassigned cells will remain set to the initial random value.

```suggestion
    /// Returns a random polynomial in the Lagrange coefficient basis,
    /// with deferred inversions.
    pub(crate) fn random_lagrange_assigned(
        &self,
        mut rng: impl RngCore,
    ) -> Polynomial<Assigned<G>, LagrangeCoeff>
    where
        G: Field,
    {
        Polynomial {
            values: (0..self.n)
                .map(|_| G::random(&mut rng).into())
                .collect(),
            _marker: PhantomData,
        }
    }
```